src/java/org/apache/cassandra/service/snapshot/TableSnapshot.java

```diff
@@ -138,11 +152,21 @@ public boolean isExpiring()
 
     public long computeSizeOnDiskBytes()
     {
-        return snapshotDirs.stream().mapToLong(FileUtils::folderSize).sum();
+        long sum = sizeOnDisk;
+        if (sum == 0)
+        {
+            for (File snapshotDir : snapshotDirs)
+                sizeOnDisk = sum += FileUtils.folderSize(snapshotDir);
+        }
```

Comment:
I'm worried about races here. What if we have two concurrent requests to the `Table Snapshot` virtual table. 2 threads will see sum == 0 and compute the sizeOnDisk twice. Should `sizeOnDisk` potentially be volatile as well? Maybe a way to prevent two threads increasing this value independently is to refactor this code as this:

```suggestion
        long sum = sizeOnDisk;
        if (sum == 0)
        if (sum == 0)
        {
            for (File snapshotDir : snapshotDirs)
                sum += FileUtils.folderSize(snapshotDir);
            sizeOnDisk = sum;
        }
```

With the only potential side effect being that two threads do the same work.

Reply:
Another potential issue is that 1 thread starts updating the `sizeOnDisk`. The second threads sees a partial, but incorrect value of the sizeOnDisk

Reply:
How can the race happen querying the vtable when the version of `getSnapshots` which is eventually called in SnapshotManager is `synchronized`?

Reply:
And yes, we can make it `volatile`.

Reply:
It think it would be the best if we just made that method `synchronized`. This method is supposed to be called just once anyway because it is not counting it all over again, just once, when sum is 0.